redfish-core/lib/assembly.hpp

```diff
@@ -367,7 +367,62 @@ inline void
 }
 
 /**
- * @brief Set location indicator for the assemblies associated to given chassis
+ * @brief API to reset fault LED.
+ * After concurrent maintenance has been performed successfully, fault LED needs
+ * to be reset.
+ * @param[in] asyncResp - Shared pointer for asynchronous calls.
+ * @param[in] assembly - Assembly object path.
+ */
+inline void
+    resetFaultLEDAfterCM(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                         const std::string& assembly)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp,
+         assembly](const boost::system::error_code ec,
+                   const std::variant<std::vector<std::string>>& resp) {
+            if (ec)
+            {
+                BMCWEB_LOG_DEBUG << "DBUS response error, ec: " << ec.value();
+                return;
+            }
+
+            const std::vector<std::string>* endpoints =
+                std::get_if<std::vector<std::string>>(&resp);
+
+            if (endpoints == nullptr)
+            {
+                BMCWEB_LOG_DEBUG << "No endpoints, skipping reset of fault LED";
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            for (auto& endpoint : *endpoints)
+            {
+                crow::connections::systemBus->async_method_call(
+                    [asyncResp, endpoint](const boost::system::error_code ec2) {
+                        if (ec2)
+                        {
+                            BMCWEB_LOG_ERROR << "Failed to reset fault LED"
+                                             << ec2;
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+                    },
+                    "xyz.openbmc_project.LED.GroupManager", endpoint,
```

Comment:
Doesn't this set all fault leds, what if you have a fault led unrelated to your CM?

Reply:
ack.